File: lib/swipe_card_event.dart
import 'dart:convert';

import 'package:cpaysdk_flutter_plugin/track_data.dart';

class SwipeCardEvent {
  String? name;
  TrackData? trackData;
  num? errorCode;

  SwipeCardEvent();

  SwipeCardEvent.fromJson(Map json) {
    name = json['name'] as String?;
    if(json['trackData']!=null){
      trackData = TrackData.fromJson(json['trackData']  as Map);
    }
    errorCode = json['errorCode'] as num?;
  }

  Map<String, dynamic> toJson() => <String,dynamic>{
        "name": this.name,
        "trackData": this.trackData,
        "errorCode": this.errorCode,
      };
}

File: android/src/main/kotlin/com/centerm/cpaysdk_flutter_plugin/SwipeCardEvent.kt
package com.centerm.cpaysdk_flutter_plugin

import com.pos.sdk.magcard.TrackData

/**
 * @Description:
 * @Author:         huangmin
 * @Date:     9/27/21 2:24 PM
 */
class SwipeCardEvent {
    var name: String? = null
    var trackData: TrackData? = null
    var errorCode :Int?=null
}